src/PersonalNumber/UK.elm

```diff
@@ -0,0 +1,346 @@
+      - let users enter upper and lower case letters, additional spaces and punctuation
+      - ignore any unwanted characters before validating
+      - avoid using ‘AB 12 34 56 C’ as an example because it belongs to a real person and use ‘QQ 12 34 56 C’ instead
+      - set the spellcheck attribute to false so that browsers do not spellcheck the National Insurance Number
+
+I've done all but the last one for you in `fromString`.
+
+
+# Definition
+
+@docs NationalInsuranceNumber
+
+
+# Strings
+
+@docs fromString, toString, display
+
+
+# Errors
+
+@docs ValidationError, errorToString, simpleError
+
+
+# JSON
+
+@docs decoder, encode
+
+-}
+
+import Json.Decode
+import Json.Encode
+
+
+{-| An opaque type representing a valid personal number.
+-}
+type NationalInsuranceNumber
+    = NINO String
+
+
+{-| If the parsing fails an error of this type is returned.
+From <https://www.gov.uk/hmrc-internal-manuals/national-insurance-manual/nim39110>
+
+A National Insurance Number is made up of 2 letters, 6 numbers and a final letter, which is always A, B, C, or D.
+
+It looks something like this:
+
+    QQ 12 34 56 A
+
+This is an example only and should not be used as an actual number.
+All prefixes are valid except:
+
+  - The characters D, F, I, Q, U, and V are not used as either the first or second letter of a NINO prefix.
+  - The letter O is not used as the second letter of a prefix.
+  - Prefixes BG, GB, KN, NK, NT, TN and ZZ are not to be used
+
+-}
+type ValidationError
+    = WrongLength -- should be 9 if spaces are removed
+    | DoesNotStartWithTwoLetters
+    | DoesNotHaveSixNumbersInTheMiddle
+    | DoesNotEndWithALetter
+    | DisallowedPrefix -- BG, GB, KN, NK, NT, TN, ZZ
+    | DisallowedPrefixLetter -- D, F, I, Q, U, V
+    | DisallowedSecondLetterO
+    | SuffixNotAtoD
+
+
+{-| Describing errors in a String.
+
+    List.map NI.errorToString
+        [ NI.DoesNotStartWithTwoLetters
+        , NI.DoesNotHaveSixNumbersInTheMiddle
+        , NI.DoesNotEndWithALetter
+        , NI.DisallowedPrefix
+        , NI.DisallowedPrefixLetter
+        , NI.DisallowedSecondLetterO
+        , NI.SuffixNotAtoD
+        ]
+        == [ "National Insurance Numbers should start with two letters, like  QQ 12 34 56 C  or  QQ123456C."
+           , "National Insurance Numbers should have six digits in the middle, like  QQ 12 34 56 C  or  QQ123456C."
+           , "National Insurance Numbers should end with a single letter A, B, C or D, like  QQ 12 34 56 C  or  QQ123456C."
+           , "BG, GB, KN, NK, NT, TN and ZZ are not allowed at the start of National Insurance Numbers."
+           , "D, F, I, Q, U, and V are not allowed in the first two letters of National Insurance Numbers."
+           , "O is not allowed in the second letter of National Insurance Numbers."
+           , "The last letter of National Insurance Numbers can only be A, B, C or D."
+           ]
+
+-}
+errorToString : ValidationError -> String
+errorToString e =
```

Comment:
This function is not needed and you should probably remove it. Most likely the user of the library will want to customise or localise the error messages, so providing built-in error message strings in not very useful and will encourage bad behaviour (e.g., not thinking about the UX).

You might consider adding a code example to the documentation instead that includes the error message strings from this function. That way a potential user can easily copy paste that code to get started with default error messages.